### PR TITLE
Mod._eval_is_integer returns a false negative

### PR DESCRIPTION
**Pull request by drufat**
When Mod._eval_is_integer cannot determine it is an integer, it should return None, and not False. 
This fixes issue #10024.

### CODE REVIEW

File: sympy/core/tests/test_arit.py [hunk 1]
```diff
@@ -1594,6 +1594,9 @@ def test_Mod():
     n = Symbol('n', odd=True)
     assert Mod(n, 2) == 1
 
+    # issue 10024
```
Comment (smichr) [inline sympy/core/tests/test_arit.py]: This should probably go at the end of `core/test_assumptions`

**Comment by drufat** [comment]
Done.

**Comment by drufat** [comment]
Fixed.

**Comment by smichr** [comment]
Thanks. It's in.

Note: in order for auto-closing of an issue to work you cannot put any other word between the action word and the issue number: not "closes issue #10024", but closes #10024.

**Pull request merged**